project-one/application.py

```diff
@@ -93,10 +93,29 @@ def book_search():
 
 
 # the book details page
-@app.route("/books/<isbn>")
-def book_details(isbn,title):
-    print(isbn, title)
-    # return render_template('book.html', )
+@app.route("/books/<isbn>" , methods = ['POST', 'GET'])
+def book_details(isbn):
+    b = Books.query.filter_by(isbn = isbn).first()
+    r = Reviews.query.filter_by(book_isbn = isbn)
+    if request.method == 'POST':
+        re = request.form.get('post-review-data')
+        re1 = request.form.get('rating-value')
+        try:
+            if re != '' and  re1 != None :
+                rev = Reviews(book_isbn = b.isbn, user_name = session.get("USERNAME"), rating = re1, review = re)
+                db.session.add(rev)
```

Comment:
where's the validation to see that the user has already entered a review or not?

Reply:
how are you sure that the reviews are added into the review table and not any other table?

Reply:
> where's the validation to see that the user has already entered a review or not?

The review table in models.py has a attribute unique for the user_name and book_isbn so when user again tries to review it results an exception so used try to except block to handle it and exception u can see the flash message.

Reply:
nice way to handle it